Update documentation about dynamic SSH key injection (#870)
kubevirt/user-guide: docs/user_workloads/accessing_virtual_machines.md

Please note that new Secrets cannot be attached to a running VM: You must restart the VM to attach the new Secret.

> **Requirement:** The qemu-guest-agent must be installed within the guest.

> > **Deprecation Notice:** The implementation supporting qemu-guest-agent versions older than 5.2 is deprecated

Reviewer: ```suggestion
!!! note "Deprecation notice" 
    Support for `qemu-guest-agent` versions older than 5.2 is deprecated and will be removed in a future release.
```
https://squidfunk.github.io/mkdocs-material/reference/admonitions/#changing-the-title

> > and will stop working in a future release.

Reviewer 2: ```suggestion
> and it will stop working in a future release.
```

> > **Deprecation Notice:** The implementation supporting qemu-guest-agent versions older than 5.2 is deprecated
> > and will stop working in a future release.

Reviewer 3: ```suggestion
> **Deprecation Notice:** Support for qemu-guest-agent versions older than 5.2 is deprecated
> and will be removed in a future release.
```

> > **Important:** When using `qemuGuestAgent` propagation,

Reviewer 2: ```suggestion
> **Important:** When using `qemuGuestAgent` propagation method,
```

> > **Important:** When using `qemuGuestAgent` propagation,

Reviewer: ```suggestion
!!! Important 
    When using `qemuGuestAgent` propagation method,
```
You'd need to replace the subsequent '>' characters and add the 4 spaces (as here) to keep it in the same admonition.
I think this also belongs above the 'requirements' heading in the opening text.

> the `/home/$USER/.ssh/authorized_keys` file is owned by the guest agent.
> Changes to the file not made by the guest agent will be lost.

> > **Further Reading:** More information about the motivation behind the access credentials API
> > can be found in the
> > [pull request description](https://github.com/kubevirt/kubevirt/pull/4195)
> > that introduced the API.

Reviewer: ```suggestion
For more information about the motivation behind the access credentials API,
see the [pull request description](https://github.com/kubevirt/kubevirt/pull/4195)
that introduced the API.
```
I would also move this above the requirements heading, and in to the opening text. (iiuc)

> In the example below, the `Secret` containing the SSH public key is

Reviewer: And then we can add an example heading here to distinguish it from the preceding section.
```suggestion
#### Example: Dynamic SSH key injection in a Secret
In this example, the `Secret` containing the SSH public key is
```
